timesketch/api/v1/resources_test.py

```diff
@@ -159,6 +159,49 @@ def test_append_label_to_sketch(self):
         )
         self.assert200(response)
 
+    def test_archive_sketch(self):
+        """Authenticated request to archive a sketch."""
+        self.login()
+
+        # Create sketch to test with
+        data = dict(
+            name="test_archive_sketch",
+            description="test_archive_sketch",
+        )
+        response = self.client.post(
+            "/api/v1/sketches/",
+            data=json.dumps(data, ensure_ascii=False),
+            content_type="application/json",
+        )
+        created_id = response.json["objects"][0]["id"]
+
+        self.assertEqual(HTTP_STATUS_CODE_CREATED, response.status_code)
+
+        # Pull sketch
+        response = self.client.get(f"/api/v1/sketches/{created_id}/")
+        self.assertEqual(HTTP_STATUS_CODE_OK, response.status_code)
+        self.assertEqual(len(response.json["objects"]), 1)
+        self.assertEqual(response.json["objects"][0]["name"], "test_archive_sketch")
+
+        # Archive sketch
+        resource_url = f"/api/v1/sketches/{created_id}/archive/"
+        data = {"action": "archive"}
+        response = self.client.post(
+            resource_url,
+            data=json.dumps(data, ensure_ascii=False),
+            content_type="application/json",
+        )
+        self.assert200(response)
+
+        # Pull the sketch again to get the status
+        response = self.client.get(f"/api/v1/sketches/{created_id}/")
+        self.assertEqual(
+            response.json["objects"][0]["name"],
+            "test_archive_sketch",
+        )
+        self.assert200(response)
+        self.assertIn("archived", response.json["objects"][0]["status"][0]["status"])
+
```

Comment:
You could add one more step in here to include testing the `unarchive` function as well. What do you think?

Reply:
yeah I was considering this, I want to check further if I want to have several things tested in the same case, usually I try to keep it separated.